src/main/java/com/google/api/codegen/transformer/SurfaceNamer.java

```diff
@@ -39,7 +39,9 @@ public SurfaceNamer(NameFormatter languageNamer) {
     super(languageNamer);
   }
 
-  public static final String NOT_IMPLEMENTED = "$ NOT IMPLEMENTED $";
+  public String getNotImplementedString(String feature) {
```

Comment:
Agreed - scanning the output and failing after the whole file has been generated is superior.